frontend/src/components/profile/info/BadgeHandler.ts

```diff
@@ -0,0 +1,252 @@
+    [BADGE.ANNONTATE_GAMES]: 'Game Annontation',
+    [BADGE.CLASSICAL_GAMES]: 'Classical Games Played',
+    [BADGE.DAILY_STREAK]: 'Daily Streak',
+};
+
+const ROUND_ROBIN_BADGES: string[] = ['fall', 'summer', 'winter', 'spring'];
+
+/**
+ * gets the RR badge
+ * @param season the RR season
+ * @returns RR image
+ */
+export function getRRbadge(season: string): string {
+    return `https://github.com/jalpp/DojoIcons/blob/main/milestones/DojoBadges/rr/clean/RR_${season}-removebg-preview.png?raw=true`;
+}
+
+/**
+ * Get the counting badge images
+ * @param level the level to fetch the badge for
+ * @param badge the type of badge
+ * @returns the image of the badge
+ */
+function getBadgeImage(level: number, badge: BADGE): string {
+    let imageURL: string;
+    switch (badge) {
+        case BADGE.POLGAR_MATE_ONE:
+            imageURL = `https://github.com/jalpp/DojoIcons/blob/main/milestones/DojoBadges/pol/new/clean/polgar_${level}-removebg-preview.png?raw=true`;
+            break;
+        case BADGE.POLGAR_MATE_TWO:
+            imageURL = `https://github.com/jalpp/DojoIcons/blob/main/milestones/DojoBadges/pol/new/clean/polgar_${level}-removebg-preview.png?raw=true`;
+            break;
+        case BADGE.POLGAR_MATE_THREE:
+            imageURL = `https://github.com/jalpp/DojoIcons/blob/main/milestones/DojoBadges/pol/new/clean/polgar_${level}-removebg-preview.png?raw=true`;
+            break;
+        case BADGE.CLASSICAL_GAMES:
+            imageURL = `https://github.com/jalpp/DojoIcons/blob/main/milestones/DojoBadges/cla/new/clean/cla_${level}-removebg-preview.png?raw=true`;
+            break;
+        case BADGE.ANNONTATE_GAMES:
+            imageURL = `https://github.com/jalpp/DojoIcons/blob/main/milestones/DojoBadges/ana/new/clean/Ann_${level}-removebg-preview.png?raw=true`;
+            break;
+        case BADGE.DAILY_STREAK:
+            imageURL = `https://github.com/jalpp/DojoIcons/blob/main/milestones/DojoBadges/streak/clean/streak${level}paper-clean.png?raw=true`;
+            break;
+    }
+
+    return imageURL;
+}
+
+/**
+ * check if a user is eligible for the badge and given requirement
+ * @param user the current user
+ * @param reqid the requirement ID
+ * @param targetLevel the target level for user to earn the badge
+ * @returns if user is eligible
+ */
+function isEligibleForBadge(user: User, reqid: string, targetLevel: number): boolean {
+    const progress = user.progress[reqid];
+    if (!progress) {
+        return false;
+    }
+
+    let totalCount = 0;
+    for (const key in progress.counts) {
+        totalCount += progress.counts[key];
+    }
+
+    return totalCount >= targetLevel;
+}
+
+/**
+ * gets the level the user is elgible for
+ * @param user the current user
+ * @param reqid the requirement id
+ * @param levels the counting badge levels list
+ * @returns the current level badge user can get
+ */
+function isEligibleForLimit(user: User, reqid: string, levels: number[]): number {
```

Comment:
This function name implies it will return a boolean but instead it returns a number